### PR TITLE
[Issue #7419] [WIP] Enable pulsar component to send their metrics to topics

### LINKED ISSUE #7419: Expose pulsar's metrics in topics
**Is your feature request related to a problem? Please describe.**
Metrics fetch is currently using prometheus endpoint that exposes metrics. External metrics collector should now poll this endpoint.

**Describe the solution you'd like**
What do you think about enable pulsar components to directly send their metrics to configuration defined topics?

Maybe we can imagine `metricstenant/brokers/<broker-name>` pattern.

### PR DESCRIPTION
**Pull request by KannarFr**
Fixes #7419

### Motivation

Enable pulsar components to send their metrics to defined topics in a dedicated tenant.

### Modifications

Create a MetricsSender interface and fill it with PulsarMetricsSender as the "to pulsar" sender.

### Verifying this change

This change added tests and can be verified as follows:

- tests are/will be defined in PulsarMetricsSenderTest

### Does this pull request potentially affect one of the following parts:

- configuration points to define interval & tenant to send.

### Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs
  - If a feature is not applicable for documentation, explain why? WIP

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/sender/PulsarMetricsSender.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    public void prepareTopics() {
+        TenantInfo tenantInfo = new TenantInfo(
+                pulsar.getConfig().getSuperUserRoles(),
+                Sets.newHashSet(pulsar.getConfig().getClusterName()));
+
+        try {
+            this.pulsar.getAdminClient().tenants().createTenant(topicToSend.getTenant(), tenantInfo);
+            this.pulsar.getAdminClient().namespaces().createNamespace(topicToSend.getNamespace());
+            this.pulsar.getAdminClient().topics().createNonPartitionedTopic(topicToSend.toString());
```
Comment (KannarFr) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/sender/PulsarMetricsSender.java:65]: @jiazhai @sijie what is the better way to manage tenant/ns/topic directly in pulsar code base? I'm currently using PulsarAdmin WDYT.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/sender/PulsarMetricsSender.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        metricsToSend.forEach(metrics -> {
+            try {
+                log.info("Sending metrics [{}]", metrics.toString());
+                this.producer.send(metrics.toString());
+            } catch (PulsarClientException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+}
```
Comment (KannarFr) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/sender/PulsarMetricsSender.java:92]: Should we embed a retry with backoff stuff?

**Comment by KannarFr** [comment]
I created the PR but it's currently a POC to get your opinions. Should we use schema using Metrics POJO?

**Review by sijie — CHANGES_REQUESTED** [review]
Why know just write an agent to collect the stats and write the stats to a Pulsar topic? It doesn't have to write such code at the broker side.

**Comment by KannarFr** [comment]
@sijie pull-based collection require a more complex infra, but also for the monitored system to buffer data during polling. So when it's under load, you'll choose between eating memory or losing data. While in push, the data is out ASAP and the collector makes that decision. So push scales much better than poll particularly in the broker case where monitoring data increases with topics number.

WDYT?

**Comment by sijie** [comment]
@KannarFr You can install a local agent per bookie to scrap the metrics from the local endpoint and push it to your metrics system. How is that different from adding a sender in Pulsar?

**Pull request closed without merging**